### ISSUE TITLE
Restructure linter flags

### ISSUE THREAD

**Issue by Hanse00** [github/super-linter#98]
**Is your feature request related to a problem? Please describe.**
As discussed in #7, we'd like to be able to run each individual linter in a separate process.
To do this, we need an easy way to enable only one specific linter per process.

**Describe the solution you'd like**
Rather than defaulting to linting all supported languages, and taking an environment variable to disable a specific language. I think we should:

Default to linting all languages (A reasonable default for the end user to expect, if they do not mess with any environment variables), but use variables to *enable* specific languages.
Meaning if no variables are present, lint all. If one variable is present, lint only that language, two variables is two languages, etc.

**Describe alternatives you've considered**
We could leave the variables to work the way they do now. However this means that if support is added for new languages, the variables passed to any one instance of the application, would have to be modified to disable the new language.

**Additional context**
Please read #7.

**Comment by Hanse00** [comment]
@admiralAwkbar What do you think of the solution I proposed in #7?

If none of the relevant env variables are present, run all linters. If any of the variables are present, default to not running any of the linters, and determine which to run based on the variables.

**Comment by admiralAwkbar** [comment]
@Hanse00 I agree and for the most part, I believe it acts in that manner, all linters are defaulted to true, unless a user sets them to false, so it wouldn't be a stretch to conform the rules to act a bit differently.

I like the idea and think its the next evolution to getting this into multiple runs.

**Comment by Hanse00** [comment]
Sounds good. I'll get to work on making it work in this manner then.

**Comment by admiralAwkbar** [comment]
@Hanse00 
![](https://media.giphy.com/media/l0ErFafpUCQTQFMSk/giphy.gif)

**Comment by Hanse00** [comment]
@admiralAwkbar I've come up with a prototype for a solution, I wanted to give you a chance to take a look before I go implementing it in the script:

```bash
#!/bin/bash

ONE=$ONE
TWO=$TWO
THREE=$THREE

VARIABLE_ARRAY=( ONE TWO THREE )

ANY_SET="false"

for i in "${VARIABLE_ARRAY[@]}"; do
    if [[ -n ${!i} ]]; then
        ANY_SET="true"
    fi
done

if [[ $ANY_SET == "true" ]]; then
    echo "One or more variables were set, defaulting the rest to false."
    for i in "${VARIABLE_ARRAY[@]}"; do
        if [[ -z ${!i} ]]; then
            eval $i="false"
        fi
    done
else
    echo "No variables were set, defaulting to true."
    for i in "${VARIABLE_ARRAY[@]}"; do
        eval $i="true"
    done
fi

for i in "${VARIABLE_ARRAY[@]}"; do
    echo "$i: ${!i}"
done
```

The above code results in the following output:
```
philip@MacBook-Pro lib % /Users/philip/Git/super-linter/lib/test.sh           
No variables were set, defaulting to true.
ONE: true
TWO: true
THREE: true
philip@MacBook-Pro lib % ONE="true" /Users/philip/Git/super-linter/lib/test.sh           
One or more variables were set, defaulting the rest to false.
ONE: true
TWO: false
THREE: false
philip@MacBook-Pro lib % ONE="true" TWO="true" /Users/philip/Git/super-linter/lib/test.sh
One or more variables were set, defaulting the rest to false.
ONE: true
TWO: true
THREE: false
```

Overall I like this solution, as it will scale well with additions / removals of possible linter options. All one needs to do is add them to the array at the start, and the rest of the looping logic stays the same.

The only concern I have is the use of the `eval` statement, which is generally considered dangerous. However it seems to be the best solution I could find in bash, to the problem of: Assign a value to a variable, which name is not known before runtime.

In this case, I think the use of `eval` is acceptable as: The right hand side is a hard coded assignment, and the left hand is the values present in the array (Not the values those variables store), thereby I don't see a way anything could be injected without the script being modified directly. But I could of course be wrong.

Let me know what you think!

**Comment by admiralAwkbar** [comment]
@Hanse00 I agree, I'm against `eval`, seems like the simple approach should do the trick. I think we can start a PR and get this going and tested. Create a PR and I will be glad to help out as well as I get time. Thanks!

**Comment by Hanse00** [comment]
This was resolved in #100 and #101, woohoo!
